### PR TITLE
Bjorncs/detect indexing mode change storeonly

### PR DESCRIPTION
**Pull request by bjorncs**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: config-model/src/main/java/com/yahoo/vespa/model/content/ContentSearchCluster.java [hunk 1]
```diff
@@ -357,6 +357,18 @@ public void handleRedundancy(Redundancy redundancy) {
         return indexedDocTypes;
     }
 
+    public List<NewDocumentType> getDocumentTypesWithStoreOnly() {
+        List<NewDocumentType> indexedDocTypes = new ArrayList<>();
```
Comment (jonmv) [inline config-model/src/main/java/com/yahoo/vespa/model/content/ContentSearchCluster.java]: Strange variabble type for a store-only set ;)

File: config-model/src/main/java/com/yahoo/vespa/model/content/ContentSearchCluster.java [hunk 2]
```diff
@@ -357,6 +357,18 @@ public void handleRedundancy(Redundancy redundancy) {
...
+        for (NewDocumentType type : documentDefinitions.values()) {
+            if (findStreamingCluster(type.getFullName().getName()).isEmpty() &&
+                    (hasIndexedCluster() && !getIndexed().hasDocumentDB(type.getFullName().getName()) ||
+                            !hasIndexedCluster())) {
```
Comment (jonmv) [inline config-model/src/main/java/com/yahoo/vespa/model/content/ContentSearchCluster.java]: Could be simplified to ```
( ! hasIndexedCluster() || ! getIndexed().hasDocumentDB(type.getFullName().getName()))
```

**Review by jonmv — APPROVED** [review]
Very nice!

File: config-model/src/main/java/com/yahoo/vespa/model/content/ContentSearchCluster.java (hunk 1 above)
Comment (bjorncs) [reply config-model/src/main/java/com/yahoo/vespa/model/content/ContentSearchCluster.java]: 😛

File: config-model/src/main/java/com/yahoo/vespa/model/content/ContentSearchCluster.java (hunk 2 above)
Comment (bjorncs) [reply config-model/src/main/java/com/yahoo/vespa/model/content/ContentSearchCluster.java]: Good point!

**Comment by kkraune** [comment]
I wonder - what is the difference between streaming and store-only? Both are really no-index - should we kill store-only (i.e. deprecate and kill in vespa-8)?

**Comment by bratseth** [comment]
We set up the search middleware if you want to do streaming. Could also be nice to know whether you plan to do streaming to tune how we do things in the content layer (although I don't think we do that today).

If we deprecate something I think it should be streaming, but due to the above I don't think we should.

**Comment by kkraune** [comment]
thanks! what _would_ be nice is a way to control the search middleware independently - say, you have a regular indexed search, but for some cases you would want to stream search the doc store (which i believe is the same in both cases). not that off if you think of it, if you want to experiment with match modes on your corpus and don't mind latency

**Pull request merged**